include/curlSingleHandler.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015, Wazuh Inc.
+ * December 27, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _CURL_SINGLE_HANDLER_HPP
+#define _CURL_SINGLE_HANDLER_HPP
+
+#include "ICURLHandler.hpp"
+#include "curl.h"
+#include "curlException.hpp"
+#include "curlHandlerType.hpp"
+#include "customDeleter.hpp"
+#include <memory>
+#include <stdexcept>
+#include <utility>
+
+using deleterCurlHandler = CustomDeleter<decltype(&curl_easy_cleanup), curl_easy_cleanup>;
+
+//! cURLSingleHandler class
+/**
+ * This class implements the ICURLHandler interface to represents a single curl handler.
+ */
+class cURLSingleHandler final : public ICURLHandler
+{
+public:
+    /**
+     * @brief Construct a new cURLSingleHandler object
+     *
+     * @param curlHandlerType Enum value of the curl handler.
+     */
+    explicit cURLSingleHandler(CurlHandlerTypeEnum curlHandlerType)
+        : ICURLHandler(curlHandlerType)
+    {
+        m_curlHandler = std::shared_ptr<CURL>(curl_easy_init(), deleterCurlHandler());
+    }
+
+    // LCOV_EXCL_START
+    ~cURLSingleHandler() override = default;
+    // LCOV_EXCL_STOP
+
+    /**
+     * @brief This method performs the request.
+     */
+    void execute() override
+    {
+        const auto resPerform {curl_easy_perform(m_curlHandler.get())};
+
+        long responseCode;
+        const auto resGetInfo = curl_easy_getinfo(m_curlHandler.get(), CURLINFO_RESPONSE_CODE, &responseCode);
```

Comment:
Brace initialization
```suggestion
        const auto resGetInfo {curl_easy_getinfo(m_curlHandler.get(), CURLINFO_RESPONSE_CODE, &responseCode)};
```

Reply:
done